Read clouds.yaml file

greg-hellings opened the pull request:
Instead of relying on environment variables, change to reading the
user's .config/openstack/clouds.yaml file for configuration options.
However, allow them to set the OS_CLOUD environment variable to change
which OpenStack cloud to connect to.

Also, tidy up some RHEL names, add RHEL 8 support, set default Fedora to
29

greg-hellings committed:
Read clouds.yaml file

Instead of relying on environment variables, change to reading the
user's .config/openstack/clouds.yaml file for configuration options.
However, allow them to set the OS_CLOUD environment variable to change
which OpenStack cloud to connect to.

Also, tidy up some RHEL names, add RHEL 8 support, set default Fedora to
29

tehsmyers commented on Vagrantfile:
```diff
@@ -103,14 +108,31 @@ end
 # then you can just run "vagrant --provider openstack up" and this will auto
 # configure auth
 Vagrant.configure("2") do |config|
+	clouds = YAML.load_file(ENV['HOME'] + '/.config/openstack/clouds.yaml')
```
There's no protection against ENOENT here, but the previous version would have similarly exploded if the env vars aren't set so I guess it's fine either way. An entry in the README  about using clouds.yml might be worthwhile.

greg-hellings replied:
The entire thing currently explodes in a glorious manner when there is no ENV['OS_*'] defined. And it explodes in a way that really doesn't help you understand what is going wrong, as there's no indication of which file or what the line is that's exploding (basically, it pulls the variables from ENV as Nil, then the OpenStack plugin itself explodes when it's trying to parse the Nil value).

However, your comment does remind me that currently it will blow a fuse even if you're not using the OpenStack provider because it won't be able to find the file. I'm going to move these lines inside the OpenStack guard to prevent that.

greg-hellings committed:
More userfriendliness

Add mapping for Fedora 29 OpenStack images
Only read clouds.yaml file when using the openstack provider
